File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer = vector(2, 0);
    int zero = 0;
    int correct = 0;
    
    for (int i = 0; i < lottos.size(); i++) {
        if (lottos[i] == 0) {
            zero++;
            continue;
        }
        auto vi = find(win_nums.begin(), win_nums.end(), lottos[i]);
        if (vi != win_nums.end()) {
            correct++;
        }
    }
    answer[0] = 7 - (correct + zero) < 6 ? 7 - (correct + zero) : 6;
    answer[1] = 7 - (correct) < 6 ? 7 - (correct) : 6;
    return answer;
}

File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    zero = lottos.count(0)
    match = 0
    
    for i in range(len(lottos)):
        if (lottos[i] != 0) and (lottos[i] in win_nums):
            match += 1
    
    high = 7 - (match + zero) if 7 - (match + zero) < 7 else 6
    low = 7 - match if 7 - match < 7 else 6
    answer = [high, low]
    return answer